dali/core/mm/mm_test_utils.h

```diff
@@ -29,18 +30,24 @@ namespace mm {
 namespace test {
 
 template <bool owning, bool security_check,
-          typename Base, typename Upstream, typename... ExtraParams>
-class test_resource_wrapper : public Base {
+          typename Base, typename Upstream>
+class test_resource_wrapper;
+
+template <bool owning, bool security_check, typename Upstream>
+class test_resource_wrapper_impl {
```

Comment:
Why you need to split wrapper into impl and wrapper itself?

Reply:
The `impl` has template callable UpstreamAlloc and UpstreamDealloc which can resolve to `upstream_->(de)allocate` or `upstream_->(de)allocate_async` (with stream). Without that I'd need to have 2 or 3 near-copies of alloc/dealloc functions.

Reply:
ok